tests/DirEnt.c

```diff
@@ -0,0 +1,49 @@
+#include <dirent.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+
+void check_error(const char *msg) {
+    if (errno) {
+        perror(msg);
+        exit(1);
+    }
+}
+
+int main() {
```

Comment:
I don't understand the purpose of this. We check for headers and library functions in `configure.ac`

https://github.com/haskell/unix/blob/eced23b356ed00510a7e6735b3f093f0c7f94081/configure.ac#L30-L66

Reply:
Explaining in a comment in the code.